Sources/SwiftFormat/Utilities/FileIterator.swift

```diff
@@ -83,6 +83,21 @@ public struct FileIterator: Sequence, IteratorProtocol {
           fallthrough
 
         case .typeDirectory:
+          do {
+            if let ignoreFile = try IgnoreFile(forDirectory: next), !ignoreFile.shouldProcess(next) {
+              // skip this directory and its subdirectories if it should be ignored
+              continue
+            }
+          } catch IgnoreFile.Error.invalidContent(let url) {
+            // we hit an invalid ignore file
+            // we skip the directory, but return the path of the ignore file
+            // so that we can report an error
+            output = url
+          } catch {
+            // we hit another unexpected error; just skip the directory
+            continue
```

Comment:
I don’t think that silently skipping a directory from formatting is the right answer here because it gives a false confidence that source files are formatted correctly. I think we should either expose the error to the user or not skip the directory.

Reply:
Ok, I've made it just carry on. It's tricky to expose the error here due to the design of FileIterator.